### PR TITLE
CinderCSI: Add cloning support

### LINKED ISSUE #644: Add Cloning support for Cinder CSI driver
CSI 1.0.0 spec supports volume cloning support . Add the support for the same in cinder CSI driver.

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
This PR adds support for cloning in cinder CSI driver
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #644

**Special notes for your reviewer**:

**Release note**:

```release-note
Volume Cloning support added for Cinder CSI driver
```

### CODE REVIEW

File: examples/cinder-csi-plugin/clone/source-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: source-pvc
+spec:
+  storageClassName: csi-cinderplugin-sc
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
```
Comment (jichenjc) [inline examples/cinder-csi-plugin/clone/source-pvc.yaml]: maybe other than those examples, you can add some info in the doc as well?

Comment (ramineni) [reply examples/cinder-csi-plugin/clone/source-pvc.yaml]: sure, Ill add in separate PR ?

Comment (jichenjc) [reply examples/cinder-csi-plugin/clone/source-pvc.yaml]: works for me ,thanks

**Comment by ramineni** [comment]
cc @adisky @lingxiankong

**Comment by ramineni** [comment]
@chrigl @lingxiankong @adisky can we have eyes on this. Its completed from my side.

**Comment by lingxiankong** [comment]
@chrigl could you please have a look cause you are more familiar with cinder-csi plugin than me?

**Pull request merged**